scripts/west_commands/sign.py

```diff
@@ -244,6 +246,26 @@ def sign(self, command, build_dir, build_conf, formats):
         '''
 
 
+class CommonSigner(Signer):
+    @staticmethod
+    def get_tool(command, tool_name):
+        if command.args.tool_path:
+            tool = command.args.tool_path
+            if not os.path.isfile(tool):
+                command.die(f'--tool-path {tool}: no such file')
+        else:
+            tool = shutil.which(tool_name)
+            if not tool:
+                command.die(f'"{tool_name}" not found; either make it available on PATH or provide --tool-path')
+        return tool
+
+    @staticmethod
+    def get_basename(build_dir, build_conf):
+        return (pathlib.Path(build_dir) / 'zephyr' /
+                         build_conf.get('CONFIG_KERNEL_BIN_NAME', "zephyr"))
```

Comment:
AFAIK, all "basename" functions strip directories. I don't think this function should return directories. Generally speaking, nothing ever combines directories + steam without suffix, that's confusing.

I checked the `pathlib` documentation and it uses the word "stem". That sounds good to me, sorry I suggested "basename" earlier.

Reply:
Could this maybe accept `.elf` and `.signed.elf` as an argument (suffix or extension) and add it within the function?